cmd/gateway-main.go

```diff
@@ -117,6 +119,31 @@ func newGatewayConfig(accessKey, secretKey, region string) error {
 	return nil
 }
 
+// Return endpoint.
+func parseGatewayEndpoint(arg string) (endPoint string, secure bool, err error) {
+	schemeSpecified, err := regexp.MatchString(".*://", endPoint)
```

Comment:
Also, for a simple check like this, it is not necessary to use regular expressions:

```
schemeSpecified := len(strings.Split(arg, "://")) > 1
```

This is faster, and there is no error checking needed.